src/app/project/project-cac/project-cac.component.ts

```diff
@@ -0,0 +1,413 @@
+    private router: Router,
+    private snackBar: MatSnackBar,
+    private excelService: ExcelService,
+    private route: ActivatedRoute,
+    private _changeDetectionRef: ChangeDetectorRef
+  ) { }
+
+  ngOnInit() {
+    this.currentProject = this.storageService.state.currentProject.data;
+
+    this.loading = true;
+
+    this.route.params.subscribe(params => {
+      //  Set default params
+      let newParams = Object.assign({}, params);
+      if (!newParams.sortBy || newParams.sortBy === '') {
+        newParams['sortBy'] = '-email';
+      }
+      this.tableParams = this.tableTemplateUtils.getParamsFromUrl(newParams, null, 25);
+    });
+
+    this.route.data
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((data: any) => {
+        if (data) {
+          if (data.project && data.project[0].data.meta && data.project[0].data.meta.length > 0) {
+            this.projectCAC = data.project[0].data.searchResults[0].projectCAC;
+          }
+          if (data.cacMembers && data.cacMembers[0].data.meta && data.cacMembers[0].data.meta.length > 0) {
+            this.tableParams.totalListItems = data.cacMembers[0].data.meta[0].searchResultsTotal;
+            this.cacMembers = data.cacMembers[0].data.searchResults;
+          } else {
+            this.tableParams.totalListItems = 0;
+            this.cacMembers = [];
+          }
+          this.setRowData();
+          this.loading = false;
+          this._changeDetectionRef.detectChanges();
+        } else {
+          alert('Uh-oh, couldn\'t load valued components');
+          // project not found --> navigate back to search
+          this.router.navigate(['/search']);
+        }
+      });
+  }
+
+  setRowData() {
+    let list = [];
+    if (this.cacMembers && this.cacMembers.length > 0) {
+      this.cacMembers.forEach(item => {
+        list.push(
+          item
+        );
+      });
+      this.tableData = new TableObject(
+        CACTableRowsComponent,
+        list,
+        this.tableParams
+      );
+    }
+  }
+
+  setColumnSort(column) {
+    if (this.tableParams.sortBy.charAt(0) === '+') {
+      this.tableParams.sortBy = '-' + column;
+    } else {
+      this.tableParams.sortBy = '+' + column;
+    }
+    this.getPaginatedDocs(this.tableParams.currentPage);
+  }
+
+  isEnabled(button) {
+    if (button === 'edit') {
+      return this.selectedCount === 1;
+    } else {
+      return this.selectedCount > 0;
+    }
+  }
+
+  updateSelectedRow(count) {
+    this.selectedCount = count;
+  }
+
+  public onPageLimitClick(pageLimit: number | string) {
+    if (pageLimit === 'all') {
+      this.tableParams.pageSize = this.tableParams.totalListItems;
+    } else {
+      this.tableParams.pageSize = pageLimit as number;
+    }
+    this.onSubmit();
+  }
+
+  async copyEmail() {
+    let itemsToExport = [];
+    this.tableData.data.map((item) => {
+      if (item.checkbox === true) {
+        itemsToExport.push(item);
+      }
+    });
+    let list = [];
```

Comment:
Anywhere we use `map` and push to an array we can `let list = itemsToExport.map(item => item.email)`

Reply:
Converted to forEach.  Fixed in 43abb05.